src/gf3d_pipeline.c

```diff
@@ -167,6 +237,181 @@ void gf3d_pipeline_render_pass_setup(Pipeline *pipe)
     }
 }
 
+Pipeline *gf3d_pipeline_basic_model_create_2D(VkDevice device,char *vertFile,char *fragFile,VkExtent2D extent,Uint32 descriptorCount)
+{
+    Pipeline *pipe;
+    VkRect2D scissor = {0};
+    VkViewport viewport = {0};
+    VkGraphicsPipelineCreateInfo pipelineInfo = {0};
+    VkPipelineLayoutCreateInfo pipelineLayoutInfo = {0};
+    VkPipelineViewportStateCreateInfo viewportState = {0};
+    VkPipelineRasterizationStateCreateInfo rasterizer = {0};
+    VkPipelineShaderStageCreateInfo shaderStages[2];
+    VkPipelineShaderStageCreateInfo vertShaderStageInfo = {0};
+    VkPipelineShaderStageCreateInfo fragShaderStageInfo = {0};
+    VkPipelineVertexInputStateCreateInfo vertexInputInfo = {0};
+    VkPipelineInputAssemblyStateCreateInfo inputAssembly = {0};
+    VkPipelineMultisampleStateCreateInfo multisampling = {0};
+    VkPipelineColorBlendAttachmentState colorBlendAttachment = {0};
+    VkPipelineColorBlendStateCreateInfo colorBlending = {0};
+    VkPipelineDepthStencilStateCreateInfo depthStencil = {0};
+    
+    slog("creating 2d pipeline");
+
+    pipe = gf3d_pipeline_new();
+    if (!pipe)return NULL;
+
+    pipe->vertShader = gf3d_shaders_load_data(vertFile, &pipe->vertSize);
+    pipe->fragShader = gf3d_shaders_load_data(fragFile, &pipe->fragSize);
+
+    pipe->vertModule = gf3d_shaders_create_module(pipe->vertShader, pipe->vertSize, device);
+    pipe->fragModule = gf3d_shaders_create_module(pipe->fragShader, pipe->fragSize, device);
+
+    pipe->device = device;
+    pipe->descriptorSetCount = descriptorCount;
+
+    depthStencil.sType = VK_STRUCTURE_TYPE_PIPELINE_DEPTH_STENCIL_STATE_CREATE_INFO;
+    // depthStencil.depthTestEnable = VK_FALSE;
+    depthStencil.depthTestEnable = VK_TRUE;
+    // depthStencil.depthWriteEnable = VK_FALSE;
+    depthStencil.depthWriteEnable = VK_TRUE;
+    depthStencil.depthCompareOp = VK_COMPARE_OP_LESS_OR_EQUAL;
+    depthStencil.depthBoundsTestEnable = VK_FALSE;
+    depthStencil.minDepthBounds = 0.0f; // Optional
+    depthStencil.maxDepthBounds = 1.0f; // Optional
+    depthStencil.stencilTestEnable = VK_TRUE;
+    // depthStencil.stencilTestEnable = VK_FALSE;
+
+    vertShaderStageInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+    vertShaderStageInfo.stage = VK_SHADER_STAGE_VERTEX_BIT;
+    vertShaderStageInfo.module = pipe->vertModule;
+    vertShaderStageInfo.pName = "main";
+
+    fragShaderStageInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+    fragShaderStageInfo.stage = VK_SHADER_STAGE_FRAGMENT_BIT;
+    fragShaderStageInfo.module = pipe->fragModule;
+    fragShaderStageInfo.pName = "main";
+
+    shaderStages[0] = vertShaderStageInfo;
+    shaderStages[1] = fragShaderStageInfo;
+
+    vertexInputInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_VERTEX_INPUT_STATE_CREATE_INFO;
```

Comment:
Are you attaching a vertex buffer as an input?

Reply:
Yes. I know it's not required, but I'm planning on using that to control where the gui is rendered on screen

Reply:
If you setup the vetextInputInfo and attach that to the pipeline/descriptorset then you need to attach a vertex buffer to the pipeline. I think that's what the spec says?